Userland/Applications/SoundPlayer/main.cpp

```diff
@@ -125,7 +129,21 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     visualization_actions.add_action(samples);
 
     auto album_cover_visualization = GUI::Action::create_checkable("&Album Cover", [&](auto&) {
-        static_cast<SoundPlayerWidgetAdvancedView*>(player)->set_visualization<AlbumCoverVisualizationWidget>();
+        auto get_image_from_music_file = [&player, &decoder_client]() -> RefPtr<Gfx::Bitmap> {
+            auto const& pictures = player->pictures();
```

Comment:
You're grabbing the pictures by const reference here, but this will actually always result in a copy.

`Vector<Audio::PictureData> pictures() const;` should be `Vector<Audio::PictureData> const& pictures() const;` if you want to avoid copying every single picture each time you swap tracks.

Reply:
It appears by making the function virtual for all loader on @kleines's request.
Solved the issue by adding the `Vector` as a member of `LoaderPlugin`.